src/api/procedures/moveFunds.ts

```diff
@@ -26,6 +26,25 @@ export async function checkArgs(
   const sendingAccount = asConfidentialAccount(from, context);
   const receivingAccount = asConfidentialAccount(to, context);
 
+  const [sendingAccountExists, receivingAccountExists] = await Promise.all([
+    sendingAccount.exists(),
+    receivingAccount.exists(),
+  ]);
+
+  if (!sendingAccountExists) {
+    throw new PolymeshError({
+      code: ErrorCode.DataUnavailable,
+      message: 'The sending Confidential Account does not exist',
+    });
+  }
+
+  if (!receivingAccountExists) {
+    throw new PolymeshError({
+      code: ErrorCode.DataUnavailable,
+      message: 'The receiving Confidential Account does not exist',
```

Comment:
```suggestion
      message: 'The receiving Confidential Account does not exist',
      data: { receivingAccount: receivingAccount.publicKey }
```

Same with the sender account error.

If the error happens its nice to have the actual account.